lib/ddtrace/span.rb

```diff
@@ -31,6 +31,9 @@ class Span
     # parsing 64-bit integers for distributed tracing since an upstream system may generate one
     EXTERNAL_MAX_ID = 2**64
 
+    # This limit is for numeric tags because uint64 could end up rounded.
+    MAX_NUMERIC_TAG_SIZE = 2**53
```

Comment:
Does this limit also apply to floating-point numbers greater than 2⁵³? The comment makes me think that this is only a problem for integer numbers.

Reply:
Interestingly enough:

```
irb(main):005:0> ((2**53)+0.5) > 2**53
=> false
irb(main):006:0> ((2**53)+1.5) > 2**53
=> true
```

Reply:
Indeed this limit is scoped to integer as they're exact numbers. For floats (which are already stored as approximation), we can (and should?) still put them in `Metrics` field, as there is no loss of precision involved. Sorry for the lack of precision here.

Reply:
Okay, so all numbers are stored as metrics unless its an integer greater than 2^53.